Mathlib/NumberTheory/NumberField/Embeddings.lean

```diff
@@ -1066,3 +1069,29 @@ theorem nrRealPlaces_eq_zero_of_two_lt (hk : 2 < k) (hζ : IsPrimitiveRoot ζ k)
     linarith
 
 end IsPrimitiveRoot
+
+/-!
+
+## The infinite place of the rationals.
+
+-/
+
+namespace Rat
+
+open NumberField
+
+noncomputable def infinitePlace : InfinitePlace ℚ := .mk (Rat.castHom _)
+
+instance : Subsingleton (InfinitePlace ℚ) where
+  allEq a b := Subtype.ext <| by
+    obtain ⟨_, _, rfl⟩ := a
+    obtain ⟨_, _, rfl⟩ := b
+    ext
+    simp
```

Comment:
If you add an appropriate ext lemma and tag `infinitePlace_apply` as simp then this is just `by ext; simp` or `by aesop`.

Reply:
Should `infinitePlace_apply` be `simp`? I added the `ext` lemma.